src/App.tsx

```diff
@@ -15,48 +21,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FIELD_NAME: SortType = SortType.Name;
+const SORT_FIELD_LENGTH: SortType = SortType.Length;
+
+function getPrepareGoods(goods: string[], sortField: SortType, reverse: boolean) {
+  let prepareGoods = [...goods];
+
+  if (sortField) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    prepareGoods = prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = React.useState(SortType.Defoult);
+  const [reverse, setReverse] = React.useState(false);
```

Comment:
```suggestion
  const [sortField, setSortField] = useState(SortType.Defoult);
  const [isReversed, setIsReversed] = useState(false);
```
- import useState directly from react.
`import { useState } from 'react'`

- follow [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig) in naming boolean variables